### ISSUE TITLE
CheckboxGroup gives IllegalArgumentException when renavigating in a disabled container

### ISSUE THREAD

**Issue by mehdi-vaadin** [vaadin/vaadin-checkbox-flow#81]
The following piece of code ends up in an `IllegalArgumentException`.
```
@Route("cbg")
@RouteAlias("")
public class TestCheckboxGroup extends Div {
    public TestCheckboxGroup() {
        setEnabled(false);

        CheckboxGroup<String> checkboxGroup = new CheckboxGroup<>();
        checkboxGroup.setItems("a", "b", "c");
        checkboxGroup.setValue(Collections.singleton("b"));
        add(checkboxGroup);

        UI.getCurrent().navigate(TestCheckboxGroup.class);
    }
}
```
Combination of `RouteAlias`, `setEnabled(false)` and, `navigate(TestCheckboxGroup.class)` causes this exception. Eliminating any of them is enough to avoid the exception. It happened first in bookstore-starter-flow project. It can be seen here https://github.com/vaadin/bookstore-starter-flow/tree/061d25944032b9b710b21aa5418585f378f3971e . To avoid the exception in bookstore, we commented the `setEnabled(false)` statement for now. 
https://github.com/vaadin/bookstore-starter-flow/blob/master/bookstore-starter-flow-ui/src/main/java/com/vaadin/samples/crud/SampleCrudView.java#L130

This is the stack trace:
```
There was an exception while trying to navigate to '' with the exception message 'Not in the list'
java.lang.IllegalArgumentException: Not in the list
	at com.vaadin.flow.dom.impl.AbstractNodeStateProvider.removeChild(AbstractNodeStateProvider.java:123)
	at com.vaadin.flow.dom.Node.removeChild(Node.java:332)
	at com.vaadin.flow.dom.Element.removeFromParent(Element.java:568)
	at com.vaadin.flow.dom.Node.insertChild(Node.java:249)
	at com.vaadin.flow.dom.Node.appendChild(Node.java:141)
	at com.vaadin.flow.dom.Element.setTextContent(Element.java:992)
	at com.vaadin.flow.dom.Element.setText(Element.java:977)
	at com.vaadin.flow.component.checkbox.Checkbox.setLabel(Checkbox.java:120)
	at com.vaadin.flow.component.checkbox.CheckboxGroup.updateCheckbox(CheckboxGroup.java:324)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.stream.SpinedBuffer$1Splitr.forEachRemaining(SpinedBuffer.java:364)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at com.vaadin.flow.component.checkbox.CheckboxGroup.refreshCheckboxes(CheckboxGroup.java:307)
	at com.vaadin.flow.component.checkbox.CheckboxGroup.onEnabledStateChanged(CheckboxGroup.java:146)
	at com.vaadin.flow.component.ComponentUtil.onComponentDetach(ComponentUtil.java:277)
	at java.base/java.util.Optional.ifPresent(Optional.java:183)
	at com.vaadin.flow.internal.nodefeature.ComponentMapping.onDetach(ComponentMapping.java:107)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at com.vaadin.flow.internal.StateNode.forEachFeature(StateNode.java:343)
	at com.vaadin.flow.internal.StateNode.fireDetachListeners(StateNode.java:769)
	at com.vaadin.flow.internal.StateNode.onDetach(StateNode.java:333)
	at com.vaadin.flow.internal.StateNode.setParent(StateNode.java:272)
	at com.vaadin.flow.internal.nodefeature.NodeFeature.detatchPotentialChild(NodeFeature.java:95)
	at com.vaadin.flow.internal.nodefeature.StateNodeNodeList.remove(StateNodeNodeList.java:65)
	at com.vaadin.flow.internal.nodefeature.ElementChildrenList.remove(ElementChildrenList.java:52)
	at com.vaadin.flow.dom.impl.AbstractNodeStateProvider.removeChild(AbstractNodeStateProvider.java:125)
	at com.vaadin.flow.dom.Node.removeChild(Node.java:332)
	at com.vaadin.flow.dom.Element.removeFromParent(Element.java:568)
	at com.vaadin.flow.component.internal.UIInternals.showRouteTarget(UIInternals.java:681)
	at com.vaadin.flow.router.internal.AbstractNavigationStateRenderer.handle(AbstractNavigationStateRenderer.java:204)
	at com.vaadin.flow.router.Router.handleNavigation(Router.java:221)
	at com.vaadin.flow.router.Router.navigate(Router.java:192)
	at com.vaadin.flow.router.Router.initializeUI(Router.java:95)
	at com.vaadin.flow.server.BootstrapHandler.createAndInitUI(BootstrapHandler.java:1117)
	at com.vaadin.flow.server.BootstrapHandler.synchronizedHandleRequest(BootstrapHandler.java:396)
	at com.vaadin.flow.server.SynchronizedRequestHandler.handleRequest(SynchronizedRequestHandler.java:40)
	at com.vaadin.flow.server.VaadinService.handleRequest(VaadinService.java:1533)
	at com.vaadin.flow.server.VaadinServlet.service(VaadinServlet.java:227)
```

**Comment by KasparScherrer** [comment]
Hi
I am experiencing this issue as well. However, I do not have a `@RouteAlias`! 
When I navigate away from the view, the same stack trace is shown.

Simply commenting the `setEnabled(false)` of the component like you did in the bookstore starter project is not an option for me. I will need to hide it as a workaround, but this will create layouting issues in my view which I then need to address as well, unnecessarily.

**Comment by pekam** [comment]
Fix was released in `vaadin-checkbox-flow 1.3.2`, which will be included in the upcoming Vaadin platform version `13.0.5`
